### PR TITLE
1262 | Enrich Documentation for SupernovaI2CBlockingInterface Class

### PR DESCRIPTION
**Pull request by gusaaaaa**
**Resolves**

- [1262 | Enrich Documentation for SupernovaI2CBlockingInterface Class](https://focusuy.atlassian.net/browse/BMC2-1262)

**How to test**

- Read the docstring and make sure that it make sense and is aligned with the ticket's acceptance criteria.

### CODE REVIEW

File: supernovacontroller/sequential/i2c.py [hunk 1]
```diff
@@ -5,6 +5,62 @@
...
+    This class includes three primary methods:
+
+    - write(target_static_address, register_address, data):
+        Writes data to the specified I2C target device.
+
+    - read(target_static_address, length):
+        Reads data from the specified I2C target device.
+
+    - read_from(target_static_address, register_address, length):
+        Reads data from the specified I2C target device.
```
Comment (focus-martin) [inline supernovacontroller/sequential/i2c.py:20]: There is room to enhance this comment so it is not the same comment as the description of the `read` method. My suggestion below:

_Reads data from the specified I2C target device, indicating the address of the internal register from which the data is read._

File: supernovacontroller/sequential/i2c.py [hunk 2]
```diff
@@ -5,6 +5,62 @@
...
+          This value is optional because it depends on the target. Most I2C targets provide a memory register map so the
+          I2C controller can access and write to or read from the memory by indexing it with a register address. However,
+          some I2C targets receive bytes that are interpreted as commands. If the internal register is not used, the
+          register address list can be left empty. For example: i2c.write(0x18, [], [0xF1, 0xF2, 0xF3, 0xF4, 0xF5]).
+          The maximum length of the register address list is 4. For example: i2c.write(0x18, [0x01, 0x02, 0x03, 0x04],
+          [0xF1, 0xF2, 0xF3, 0xF4, 0xF5]).
+        - data: A Python list of bytes holding the data to be sent over I2C. The maximum length is 1024 bytes.
+
+        When passing both lists (register_address and data) populated with data, the stream of bytes transferred over
+        I2C is the addition of both lists. For instance, if you send the command i2c.write(0x18, [0x01, 0x02, 0x03, 0x04],
```
Comment (focus-martin) [inline supernovacontroller/sequential/i2c.py:38]: I would change _For instance, if you send the command i2c.write_ to _For instance, if the user sends the command i2c.write_

File: supernovacontroller/sequential/i2c.py [hunk 3]
```diff
@@ -5,6 +5,62 @@
...
+
+        Reads data from the I2C target device without specifying a register address.
+
+    i2c.read_from(target_static_address, register_address, length):
+        - target_static_address: This is the static address of the I2C target.
+        - register_address: A Python list representing the register address of the internal memory of the I2C target.
+          This value is optional and depends on the target. The maximum length of the register address list is 4.
+        - length: The number of bytes to be read from the target.
+
+        The description of the arguments target_static_address and register_address above applies to this method as well.
```
Comment (focus-martin) [inline supernovacontroller/sequential/i2c.py:54]: Maybe this sentence doesn't make sense?

**Review by focus-martin — CHANGES_REQUESTED** [review]
Gus, thank you for embracing this changes after the conversation feedback sent to the customer. I left some comment, check them and let me know what do you think. 

Finally, what do you think about enriching the docstrings documentation of the method instead of only enriching the docstring of the class?

File: supernovacontroller/sequential/i2c.py (hunk 2 above)
Comment (gusaaaaa) [reply supernovacontroller/sequential/i2c.py:38]: You are right! I've decided to use the same format as the previous sentence :)

File: supernovacontroller/sequential/i2c.py [hunk 4]
```diff
@@ -5,6 +5,61 @@
...
+        - target_static_address: This is the static address of the I2C target.
+        - register_address: A Python list representing the register address of the internal memory of the I2C target.
+          This value is optional and depends on the target. The maximum length of the register address list is 4.
+        - length: The number of bytes to be read from the target.
+
+        The length parameter specifies the number of bytes to be read from the target.
+
+    Example Use Case:
+    -----------------
+    - When writing to an I2C target that uses an internal memory register map, ensure that the register address length
```
Comment (Salonso1602) [inline supernovacontroller/sequential/i2c.py:58]: This is more of a "Clarification" or "Note" rather than an example use case

**Review by Salonso1602 — CHANGES_REQUESTED** [review]
I have left a minor comment regarding the doc. I also subscribe to Martin's suggestion, maybe you can just keep this as is and add the subsections of this class annotations to the specific methods it refers.
What do you think?

File: supernovacontroller/sequential/i2c.py (hunk 4 above)
Comment (gusaaaaa) [reply supernovacontroller/sequential/i2c.py:58]: Good catch, thanks.

**Pull request merged**